### DISCUSSION TITLE
Adding Sleep Stage or Publisher Stage

### DISCUSSION THREAD

**Discussion by mrceki** [ros-planning/moveit_task_constructor#636] [Q&A] [answered]
Hi,

I am wondering if it is possible to add a sleep stage or some publisher stage to the task constructor. If it is possible, what is the best way to implement it?

Thank you in advance!

**Comment by rhaschke** [comment, marked as answer]
I guess you intend to pause execution, not planning. This is not possible by design: MTC stages's code is executed during planning only. The resulting trajectory is just passed to MoveIt's execution manager.
The same probably holds for a publisher stage. (Not sure what you want to publish.)

**Reply by mrceki** [reply]
Yes, I want to pause execution and I was actually aiming to adjust the control parameters of the manipulator by dynamic reconfigure between stages, but I couldn't find a way to do it. Do you have any suggestions?

**Reply by rhaschke** [reply]
Configure different controller parameter sets as different named controllers, which you can then refer to in a `TrajectoryExecutionInfo` message passed to `Stage::setTrajectoryExecutionInfo`
https://github.com/moveit/moveit_task_constructor/blob/d5e1376b08450391f80e74eb51070cf1f1ae1453/msgs/msg/TrajectoryExecutionInfo.msg#L2
The listed controllers are then used to execute the trajectory generated by a particular stage.

**Reply by mrceki** [reply]
I meant, for example, an impedance controller, which has dynamic reconfigure capability and is capable of adjusting damping and stiffness dynamically. In general, what I want to do is adjust the stiffness and damping parameters between stages.

**Reply by rhaschke** [reply]
If you want to fiddle with parameters interactively, I suggest splitting trajectory execution manually.

**Reply by mrceki** [reply]
Thanks for your response!
